components/Navigation.tsx

```diff
@@ -127,78 +128,82 @@ export default function Navigation() {
+                  value={searchValue ?? ""}
+                  onChange={handleChange}
+                  onKeyDown={handleKeyPress}
+                  placeholder="Search..."
+                  type="text"
+                  className="m-5 drop-shadow-md rounded-sm pl-3"
+                />
+                <div className="absolute">
+                  {searchValue &&
+                    currentCoins?.filter((coin) => {
+                      const name = coin.name.toLowerCase();
+                      const search = searchValue.toLowerCase();
+                      if (name.startsWith(search))
+                        return (
+                          <div key={coin.id} className="border-slate-300">
+                            <DropdownRow
+                              key={coin.id}
+                              className="bg-second"
+                              onClick={() => handleSearch(coin.id)}
+                            >
+                              {coin.name}
+                            </DropdownRow>
+                          </div>
+                        );
+                    })}
+                </div>
 
-              <select
-                onChange={(e) => handlePalette(e)}
-                name="palette"
-                className="m-5 drop-shadow-md rounded-sm "
-              >
-                <option>Theme</option>
-                {palettes?.map((theme) => {
-                  return (
-                    <option key={theme} value={theme}>
-                      {theme}
-                    </option>
-                  );
-                })}
-              </select>
-              <div className="h-5 ">
-                <button
-                  className="bg-accent m-2"
-                  value="dark"
-                  onClick={handleMode}
+                <select
+                  onChange={(e) => handleCurrency(e)}
+                  name="currency"
+                  className="mr-5 my-5 drop-shadow-md rounded-sm "
                 >
-                  dark
-                </button>
-                <button
-                  value="light"
-                  className="bg-accent m-2"
-                  onClick={handleMode}
+                  <option>here</option>
+                  {CurrencyArray?.map((currency) => {
+                    return (
+                      <option key={currency} value={currency}>
+                        {currency}
+                      </option>
+                    );
+                  })}
+                </select>
+
+                <select
+                  onChange={(e) => handlePalette(e)}
+                  name="palette"
+                  className="mr-5 my-5  drop-shadow-md rounded-sm "
                 >
-                  light
-                </button>
+                  <option>Theme</option>
+                  {palettes?.map((theme) => {
+                    return (
+                      <option key={theme} value={theme}>
```

Comment:
forgot to add the key

Reply:
oh, yes you;re right, but would there be a possibility for that to happen?

Reply:
Maybe we are talking about two different parts. For me, there is a key which is the {theme} on line 183. Can you clarify what you are referring to?